redis/redis.go

```diff
@@ -213,61 +227,77 @@ func (c *Client) publishMessageFromBuffer(s *bufferMessage) {
 		})
 
 		if mErr == nil {
-			c.client.Publish(c.ctx, SyncChannelName, binMsg)
+			c.client.Publish(ctx, SyncChannelName, binMsg)
 		}
 
-		c.client.Set(c.ctx,
+		c.client.Set(ctx,
 			prefixKey(s.Key),
 			binRes,
 			c.getTTL(origRes))
 	}
 }
 
-func (c *Client) processReceivedMessage(msg *redis.Message) {
+func (c *Client) processReceivedMessage(ctx context.Context, msg *redis.Message) {
+	select {
+	case <-ctx.Done():
+		return
+	default:
+	}
+
```

Comment:
I think we can skip this check: worse case we'll notice the context is done after parsing the json but we're not doing anything potentially blocking:
```suggestion
```

Reply:
And we get rid of the coverage warning :)